runbotics/runbotics-orchestrator-ui/src/views/process/ProcessBrowseView/ProcessList/ProcessList.tsx

```diff
@@ -49,19 +49,30 @@ const ProcessList: VFC = () => {
                     const newPageSize = mode === ProcessListDisplayMode.GRID
                         ? DefaultPageSize.GRID : DefaultPageSize.TABLE;
                     setPageSize(newPageSize);
-                    setDisplayMode(mode);
+                    if (mode) setDisplayMode(mode);
                     history.replace(getSearchParams({
                         page, pageSize: newPageSize, search, searchField,
                     }));
+                    clearSearch();
                 }}
             />
             <ProcessPageProvider {...{
                 pageSize, setPageSize, search, searchField, page, setPage
             }}
             >
-                {displayMode === ProcessListDisplayMode.GRID
-                    ? <GridView />
-                    : <ProcessTable onAdvancedSearchChange={handleAdvancedSearch} />}
+                <If condition={displayMode === ProcessListDisplayMode.GRID}>
+                    <If condition={loading}>
```

Comment:
you can merge this 2 If components, just use else prop

Reply:
done